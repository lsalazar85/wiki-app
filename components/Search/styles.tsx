import styled from "styled-components";
import colors from '../../styles/colors';

export const SearchContainer = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;

  @media(min-width: 320px){
    margin-bottom: 1rem;
  }
  
  @media(min-width: 768px){
    margin-bottom: 3rem;
    padding: 1rem 0;
  }
`

export const SearchInput = styled.input`
  border-radius: 0.5rem;
  height: 2.5rem;
  padding: 1rem;
  outline: none;
  font-size: 1rem;
  font-weight: 400;
  text-align: center;
  margin-right: 1rem;
  border: 3px solid ${colors.SalmonSecondary};
  margin-bottom: 1rem;
  
  @media(min-width: 320px){
    width: 70%;
  }

  @media(min-width: 767px){
    width: 15rem;
  }
`
