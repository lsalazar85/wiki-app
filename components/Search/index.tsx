import { useState, SyntheticEvent } from 'react';
import { ChangeEvent } from 'react';
import { useRouter } from 'next/router';

import ButtonForm from '../Button/ButtonForm';

import { SearchContainer, SearchInput } from './styles';


const Search = () => {
  const [inputValue, setInputValue] = useState<string>('');
  const router = useRouter();

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
    event.preventDefault();
    router.push(`/character/${inputValue}`);
    setInputValue('');
  };

  return (
    <>
      <SearchContainer onSubmit={handleSubmit}>
        <SearchInput
          onChange={handleChange}
          type="text"
          placeholder="Search by name"
          data-testid="search"
          value={inputValue}
        />
        <ButtonForm text='Search'/>
      </SearchContainer>
    </>
  );
};

export default Search;
