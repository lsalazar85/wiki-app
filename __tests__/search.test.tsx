/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react'
import Search from "../components/Search";

describe('Search', () => {
    it('should render successfully', () => {
        const { asFragment } = render(
            <Search onChange={() => jest.fn()} />
        )

        expect(asFragment()).toMatchSnapshot();
        expect(asFragment()).toBeDefined()
    })

    it('should placeholder text ok', () => {
        render(<Search onChange={() => jest.fn()} />)

        const inputPlaceHolder = screen.getByPlaceholderText(/search character by name/i)

        expect(inputPlaceHolder).toBeInTheDocument()
        expect(inputPlaceHolder).toMatchSnapshot()
    })
})