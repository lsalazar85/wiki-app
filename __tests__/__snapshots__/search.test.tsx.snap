// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should placeholder text ok 1`] = `
<input
  class="styles__SearchInput-sc-1xiurla-1 dQGEEo"
  data-testid="search"
  placeholder="Search character by name"
  type="text"
  value=""
/>
`;

exports[`Search should render successfully 1`] = `
<DocumentFragment>
  <form
    class="styles__SearchContainer-sc-1xiurla-0 cSNHGB"
  >
    <input
      class="styles__SearchInput-sc-1xiurla-1 dQGEEo"
      data-testid="search"
      placeholder="Search character by name"
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
